src/allotropy/parsers/cfxmaestro/cfxmaestro_structure.py

```diff
@@ -41,11 +43,32 @@ def create_measurement_group(
     well_data: list[SeriesData],
     plate_well_count: int,
 ) -> MeasurementGroup:
-    return MeasurementGroup(
-        plate_well_count=plate_well_count,
-        measurements=[
-            Measurement(
-                # Measurement metadata
+    measurements = []
+    for data in well_data:
+        sample_doc_custom_data = data.get_custom_keys(
+            set(constants.SAMPLE_DOCUMENT_CUSTOM_KEYS)
+        )
+        device_doc_custom_data = data.get_custom_keys(
+            set(constants.DEVICE_CONTROL_DOCUMENT_CUSTOM_KEYS)
+        )
+        processed_data_doc_custom_data = data.get_custom_keys(
+            set(constants.PROCESSED_DATA_DOCUMENT_CUSTOM_KEYS)
+        )
+        # these fields are not need or are already in the asm
+        data.mark_read(
+            {
+                "Target",
+                "Biological Set Name",
+                "Content",
+                "Fluor",
+                "Cq Mean",
+                "Unnamed: 0",
+            }
+        )
+        if data.get(str, "Sample", validate=SeriesData.NOT_NAN) or data.get(
```

Comment:
You can put this at the top, and just call data.get_unread() to clear the warning if the condition is False:
```suggestion
        if data.get(str, "Sample", validate=SeriesData.NOT_NAN) or data.get(float, "Cq", validate=SeriesData.NOT_NAN):
            data.get_unread()
            continue
```